jormungandr/src/network/p2p/comm.rs

```diff
@@ -611,40 +607,38 @@ impl Peers {
         })
     }
 
-    pub fn propagate_gossip_to(
-        &self,
-        target: Id,
-        gossip: Gossip<NodeData>,
-    ) -> impl Future<Item = (), Error = Gossip<NodeData>> {
+    pub async fn propagate_gossip_to(&self, target: Id, gossip: Gossip) -> Result<(), Gossip> {
         debug!(
             self.logger,
             "sending gossip";
             "node_id" => %target,
         );
         let logger = self.logger.clone();
-        self.inner().and_then(move |mut map| {
-            if let Some(mut entry) = map.entry(target) {
-                let res = match entry.update_comm_status() {
-                    CommStatus::Established(comms) => comms.try_send_gossip(gossip),
-                    CommStatus::Connecting(comms) => {
-                        comms.set_pending_gossip(gossip);
-                        Ok(())
-                    }
-                };
-                res.map_err(|e| {
-                    debug!(
-                        logger,
-                        "gossip propagation to peer failed, unsubscribing peer";
-                        "node_id" => %target,
-                        "reason" => %e.kind(),
-                    );
-                    entry.remove();
-                    e.into_item()
-                })
-            } else {
-                Err(gossip)
-            }
-        })
+        self.inner()
+            .and_then(move |mut map| {
```

Comment:
can't we do the other way around:

```rust
   let mut map = self.inner().await?;
   if let Some(mut entry) = map.entry(target) {
      ...
   }
```

Reply:
I just find it is easier to read (separate the async logic from the processing logic

Reply:
You are right, changed in another commit.